src/pool.rs

```diff
@@ -281,6 +281,12 @@ impl<Conn: Connection> PoolInner<Conn> {
                 }
             };
 
+            // Cancel safety: All branches of this select! statement are
+            // cancel-safe (mpsc::Receiver::recv, tokio::time::sleep_until,
+            // monitoring the tokio::sync::watch::Receivers)
+            //
+            // Futurelock safety: All select arms are queried concurrently. No
+            // awaiting happens outside this concurrent polling.
```

Comment:
FWIW, this select is also futurelock-safe since it contains no arms where the future is an `&mut future`, so there is no un-dropped future that may have acquired a lock or similar resource. It might be worth mentioning that here, as well; I think part of the value of these "safety" comments is that they enumerate all of the things which, if someone were to _change_, they might introduce a problem.